pallets/parachain-staking/src/lib.rs

```diff
@@ -843,30 +858,43 @@ pub mod pallet {
 	#[pallet::hooks]
 	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
 		fn on_initialize(n: T::BlockNumber) -> Weight {
-			let mut round = <Round<T>>::get();
-			if round.should_update(n) {
-				// mutate round
-				round.update(n);
+			let current_round = <Round<T>>::get();
+			if current_round.should_run_election(n) {
+				let election_result = Self::select_top_candidates(current_round.current);
+				QueuedElectionResult::<T>::put(election_result);
+			}
+
+			if current_round.should_update(n) {
+				let election_result = match QueuedElectionResult::<T>::get() {
+					Some(result) => result,
+					None => panic!("We're hosed. There is no election result, and Joshy didn't bother to write proper error handling! //TODO"),
```

Comment:
I'd even prefer keeping the previous round results and emitting an event `NoElectionResultPanic` rather than actually panicking.